src/Components/Users/UsersListTable.js

```diff
@@ -0,0 +1,276 @@
+  },
+];
+
+const UsersListTable = () => {
+  const [order, setOrder] = useState('asc');
+  const [orderBy, setOrderBy] = useState('name');
+  const [page, setPage] = useState(0);
+  const [usersList, setUsersList] = useState(mockedUsers);
+  const rowsPerPage = 10;
+
+  const descendingComparator = (a, b, orderBy) => {
+    if (b[orderBy] < a[orderBy]) {
+      return -1;
+    }
+    if (b[orderBy] > a[orderBy]) {
+      return 1;
+    }
+
+    return 0;
+  };
+
+  const getComparator = (order, orderBy) => {
+    return order === 'desc'
+      ? (a, b) => descendingComparator(a, b, orderBy)
+      : (a, b) => -descendingComparator(a, b, orderBy);
+  };
+
+  const handleRequestSort = (event, property) => {
+    const isAsc = orderBy === property && order === 'asc';
+
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+  };
+
+  const handleChangePage = (event, newPage) => {
+    setPage(newPage);
+  };
+
+  const createSortHandler = (property) => (event) => {
+    handleRequestSort(event, property);
+  };
+
+  // Avoid a layout jump when reaching the last page with empty rows.
+  const emptyRows =
+    page > 0 ? Math.max(0, (1 + page) * rowsPerPage - usersList.length) : 0;
+
+  const deleteUser = (id) => {
+    const newUsersList = usersList.filter((user) => user.id !== id);
+    const total = usersList.length - 1;
+    const pages = total / rowsPerPage;
+
+    setUsersList(newUsersList);
+
+    if (page === pages && page !== 0) {
+      setPage(page - 1);
+    }
+  };
+
+  return (
+    <Container sx={{ my: '1rem' }}>
+      <Box>
+        <Paper>
+          <Toolbar>
+            <Typography
+              component="div"
+              id="tableTitle"
+              sx={{ mr: 'auto' }}
+              variant="h6">
+              Lista de Usuarios
+            </Typography>
+            <Button
+              component={Link}
+              to="/backoffice/users/create"
+              variant="contained">
+              Nuevo Usuario
+            </Button>
+          </Toolbar>
+          <TableContainer>
+            <Table
+              aria-labelledby="tableTitle"
+              size="small"
+              sx={{ minWidth: 600 }}>
+              <TableHead>
+                <TableRow>
+                  <TableCell sortDirection={orderBy === 'name' ? order : false}>
+                    <TableSortLabel
+                      active={orderBy === 'name'}
+                      direction={orderBy === 'name' ? order : 'asc'}
+                      onClick={createSortHandler('name')}>
+                      Nombre
+                      {orderBy === 'name' ? (
+                        <Box component="span" sx={visuallyHidden}>
+                          {order === 'desc'
+                            ? 'ordenado descendente'
+                            : 'ordenado ascendente'}
+                        </Box>
+                      ) : null}
+                    </TableSortLabel>
+                  </TableCell>
```

Comment:
Please create a new component for this behaviour as it could highly reused as it's behaviour is pretty common. A good name could be something like "SortableTableCell". And remember to make it reusable (for example, the 'name' property must be given by parameter or constructor).